### PR TITLE
ddl: fix checkDuplicateColumn and add test cases.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

**Comment by hicqu** [comment]
PTAL, thanks!

File: ddl/ddl_api_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/model"
+)
+
+type testFieldCaseSuite struct {
+	colDefs []*ast.ColumnDef
+}
+
+func (s *testFieldCaseSuite) SetUpSuite(c *C) {
```
Comment (shenli) [inline ddl/ddl_api_test.go]: You do not need to create your own suite.

**Comment by hicqu** [comment]
address comments

**Comment by hicqu** [comment]
Fixed. PTAL again, thanks.

File: .gitignore [hunk 2]
```diff
@@ -11,3 +11,5 @@ coverage.out
 tags
 temp_parser_file
 y.output
+
+cscope.*
```
Comment (hanfei1991) [inline .gitignore]: Why add this?

Comment (hicqu) [inline .gitignore]: I use cscope for code navigation. :)

Comment (bobotu) [inline .gitignore]: Why not put this rule into your global .gitignore file, and remove it from this file?

**Comment by hicqu** [comment]
PTAL again, thanks!

**Comment by zimulala** [comment]
Please put this test to `column_test.go`.
And the prefix of a test name must have `Test`.

**Comment by hicqu** [comment]
address comments

**Comment by zimulala** [comment]
@hicqu Did you forget to push your commit?

**Comment by hicqu** [comment]
Sorry, now it's pushed.

File: ddl/column_test.go [hunk 3]
```diff
@@ -935,3 +935,19 @@ func (s *testColumnSuite) colDefStrToFieldType(c *C, str string) *types.FieldTyp
...
+		colDef := &ast.ColumnDef{
+			Name: &ast.ColumnName{
+				Schema: model.NewCIStr("TestSchema"),
+				Table:  model.NewCIStr("TestTable"),
+				Name:   model.NewCIStr(name),
+			},
+		}
+		colDefs = append(colDefs, colDef)
+	}
+	c.Assert(checkDuplicateColumn(colDefs) != nil, Equals, true)
```
Comment (zimulala) [inline ddl/column_test.go]: `c.Assert(checkDuplicateColumn(colDefs), IsNil)` is better.

File: ddl/column_test.go [hunk 4]
```diff
@@ -935,3 +935,19 @@ func (s *testColumnSuite) colDefStrToFieldType(c *C, str string) *types.FieldTyp
 	c.Assert(err, IsNil)
 	return &col.FieldType
 }
+
+func (s *testColumnSuite) TestFieldCase(c *C) {
+	var colDefs []*ast.ColumnDef
```
Comment (ngaut) [inline ddl/column_test.go]: To reduce allocation, you should make with capacity by using length of fields.

**Comment by hicqu** [comment]
@ngaut , PTAL again, thanks!

File: ddl/column_test.go [hunk 5]
```diff
@@ -937,8 +937,8 @@ func (s *testColumnSuite) colDefStrToFieldType(c *C, str string) *types.FieldTyp
 }
 
 func (s *testColumnSuite) TestFieldCase(c *C) {
-	var colDefs [2]*ast.ColumnDef
 	var fields = []string{"field", "Field"}
+	var colDefs = make([]*ast.ColumnDef, len(fields))
 	for idx, name := range fields {
```
Comment (ngaut) [inline ddl/column_test.go]: s/idx/i/

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by hicqu** [comment]
fixed. @ngaut , PTAL again, thank you!

**Comment by zimulala** [comment]
LGTM

**Pull request merged**